cli/src/config.rs

```diff
@@ -454,6 +454,12 @@ cli_opt_struct! {
         /// claim fees.
         #[clap(long)]
         validator_vote_account : Pubkey => Pubkey::default(),
+
+        /// Try to do stake and unstake operations any time if set to true,
+        /// otherwise will try to stake/unstake only at the end of the epoch.
+        /// Defaults to false.
+        #[clap(long, value_name = "true/false")]
+        stake_unstake_any_time: bool => false,
```

Comment:
Optional suggestion: we could make an enum

```rust
enum StakeTime {
    Anytime,
    OnlyNearEpochEnd,
}
// Would also need FromStr impl
```

Then it is clear in all the places where the value is used, what the two cases are. With `stake_unstake_any_time=false`, you know that we can’t stake/unstake at just any time, but it doesn’t tell you at what times it *is* possible, you have to refer to the doc comment of the field. If you see `stake_time=StakeTime::OnlyNearEpochEnd` it is more self-descriptive.

But this is a minor thing really, I’m also fine with `stake_unstake_any_time`.

Reply:
That's a good idea, thanks 👍